Fixes VSTS Bug 1009809: Save failed

mkrueger opened the pull request:
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809Save
failed

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809

mkrueger committed:
Fixes VSTS Bug 1009809:
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809Save
failed

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809

Therzok commented on main/src/core/MonoDevelop.Ide/MonoDevelop.Ide/ProjectOperations.cs:
```diff
@@ -426,6 +426,8 @@ async Task SaveAsyncInternal (SolutionItem entry)
 				await entry.SaveAsync (monitor);
 				monitor.ReportSuccess (GettextCatalog.GetString ("Project saved."));
+			} catch (OperationCanceledException) {
+				monitor.Dispose ();
```
monitor.Dispose is redundant.

mkrueger replied:
fixed.

mkrueger committed:
[Ide] Handled some exceptions for bug Bug 1009809.

https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1009809Savefailed

The test scenario now works for me.

Therzok commented on main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs:
```diff
@@ -218,7 +218,12 @@ public static void SetParentToWindow (this Gtk.Window window, MonoDevelop.Compon
+				try {
+					nsWindow = parent.GetNativeWidget<NSWindow> ();
+				} catch {
+					; // nothing
```
We should at least log the exception.

mkrueger replied:
Not really it's kind of excepted and handled when nswindow == null.

Therzok commented on main/src/core/MonoDevelop.Ide/MonoDevelop.Components/Window.cs:
```diff
@@ -218,7 +218,12 @@ public static void SetParentToWindow (this Gtk.Window window, MonoDevelop.Compon
-				var nsWindow = parent.GetNativeWidget<NSWindow> ();
+				NSWindow nsWindow = null;
+				try {
+					nsWindow = parent.GetNativeWidget<NSWindow> ();
```
You need to check if the native widget is an via `parent.nativeWidget is NSWindow`. GetNativeWidget is expected to throw.

mkrueger replied:
That relies too much on the implementation details. I prefer doing it that way.

mkrueger committed:
[Core] Introduced TryGetNativeWidget method to Control.

sevoku commented:
@monojenkins rebase
